pdns/iputils.cc

```diff
@@ -436,3 +436,43 @@ bool isTCPSocketUsable(int sock)
 
   return false;
 }
+/* mission in life: parse four cases
+   1) 1.2.3.4
+   2) 1.2.3.4:5300
+   3) 2001::1
+   4) [2002::1]:53
+*/
+
+ComboAddress parseIPAndPort(const std::string& input, uint16_t port)
+{
+  if (input[0] == '[') { // case 4
+    auto both = splitField(input.substr(1), ']');
+    return ComboAddress(both.first, both.second.empty() ? port : static_cast<uint16_t>(pdns_stou(both.second.substr(1))));
+  }
+
+  string::size_type count = 0;
+  for (char c : input) {
+    if (c == ':') {
+      count++;
+    }
+    if (count > 1) {
+      break;
+    }
+  }
+  switch (count) {
+  case 0: // case 1
```

Comment:
I would seriously consider renumbering the list in the comment at the top to make this less confusing :)